crates/matrix-sdk-crypto/src/types/mod.rs

```diff
@@ -229,6 +230,121 @@ impl Serialize for Signatures {
     }
 }
 
+/// A collection of signing keys, a map from the key id to the signing key.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SigningKeys<T: Ord>(BTreeMap<T, SigningKey>);
+
+impl<T: Ord> SigningKeys<T> {
+    /// Create a new, empty, `SigningKeys` collection.
+    pub fn new() -> Self {
+        Self(BTreeMap::new())
+    }
+
+    /// Insert a `SigningKey` into the collection.
+    pub fn insert(&mut self, key_id: T, key: SigningKey) -> Option<SigningKey> {
+        self.0.insert(key_id, key)
+    }
+
+    /// Get a `SigningKey` with the given `DeviceKeyId`.
+    pub fn get(&self, key_id: &T) -> Option<&SigningKey> {
+        self.0.get(key_id)
+    }
```

Comment:
I think this method's signature should mirror the underlying map's one (using `Borrow`), maybe then the `&OwnedDeviceId` won't be needed. If it still is, I'll look into adding any missing `Borrow` implementations in Ruma.

Reply:
Right, that does indeed work: https://github.com/matrix-org/matrix-rust-sdk/pull/943/commits/7665d0bad7e09fe9ca2b299032dad928b4f338d5.